src/shogun/mathematics/linalg/LinalgBackendViennacl.h

```diff
@@ -115,6 +186,27 @@ class LinalgBackendViennaCL : public LinalgBackendGPUBase
 		return viennacl::linalg::inner_prod(a_gpu->data(a.size()), b_gpu->data(b.size()));
 	}
 
+	/** Return the largest element in the vector with ViennaCL library */
+	template <typename T>
+	T max_impl(const SGVector<T>& vec) const
+	{
+		viennacl::ocl::kernel& kernel = generate_kernel<T>();
+
+		SGVector<T> result(1);
+		result.set_const(-1);
+		GPUMemoryViennaCL<T>* result_gpu =
+			static_cast<GPUMemoryViennaCL<T>*>(linalg::to_gpu(result).gpu_vector.get());
+		GPUMemoryViennaCL<T>* vector_gpu =
+			static_cast<GPUMemoryViennaCL<T>*>(vec.gpu_vector.get());
+
+		viennacl::ocl::enqueue(kernel(vector_gpu->data(vec.size()),
+			cl_int(vec.size()), cl_int(vector_gpu->m_offset),
+			result_gpu->data(1)));
+
+		SGVector<T> result_cpu = linalg::from_gpu(new SGVector<T>(result_gpu, 1));
+		return result_cpu[0];
+	}
```

Comment:
gives error:
error for object 0x7ff2bc0b9af0: pointer being freed was not allocated
in unit-test